### PR TITLE
codespell: add codespell config, workflow, fix spelling errors.

### PR DESCRIPTION
**Pull request by klihub**
Add codespell configuration, fix existing spelling errors. Add codespell workflow to check spelling errors in pushes/PRs to/against `main`.

### CODE REVIEW

File: .github/workflows/codespell.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+jobs:
+  codespell:
+    name: Check for spelling errors
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Codespell
+        uses: codespell-project/actions-codespell@v2
```
Comment (fuweid) [inline .github/workflows/codespell.yml:18]: IMO we should remove the EOF char here

**Review by fuweid — APPROVED** [review]
LGTM

File: .github/workflows/codespell.yml (hunk 1 above)
Comment (klihub) [reply .github/workflows/codespell.yml]: Okay, I added the missing newline.

**Review by fuweid — APPROVED** [review]

**Review by mikebrow — APPROVED** [review]
LGTM

**Pull request merged**